benchmarks/bench_ml.py

```diff
@@ -79,385 +70,336 @@ def pow2intmap(x):
+    else:
+        raise ValueError("dataset not supported.")
+    return data["data"], data["target"]
+
+
+class MLBench(object):
+    """
+    A class which is used to perform benchmarking of black box optimization
+    algorithms on various machine learning problems.
+    On instantiation, the dataset is loaded that is used for experimentation
+    and is kept in memory in order to avoid reloading data.
+
+    Parameters
+    ----------
+    * `model`: scikit-learn estimator
+        An instance of a sklearn estimator.
+
+    * `dataset`: str
+        Name of the dataset.
+    """
+    def __init__(self, model, dataset):
+        X, Y = load_data_target(dataset)
+        self.X_train, self.y_train, self.X_test, self.y_test = split_normalize(
+            X, Y)
+        self.model = model
+        self.space = MODELS[model]
 
     def evaluate(self, point):
         """
-        Evaluates model on loaded dataset for particular setting of hyperparameters.
-        :param point: dict, pairs of parameter names and corresponding values
-        :return: score (more is better!) for some specific point
+        Fits model using the particular setting of hyperparameters and
+        evaluates the model validation data.
+
+        Parameters
+        ----------
+        * `point`: dict
+            A mapping of parameter names to the corresponding values
+
+        Returns
+        -------
+        * `score`: float
+            Score (more is better!) for some specific point
         """
-
-        X_train, y_train, X_test, y_test = self.dataset # train, test split of the data
+        X_train, y_train, X_test, y_test = (
+            self.X_train, self.y_train, self.X_test, self.y_test)
 
         # apply transformation to model parameters, for example exp transformation
         point_mapped = {}
-
-        for param in point.keys():
-            v = point[param]
-            v = self.model_description[MODEL_PARAMETERS][param][1](v)
-            point_mapped[param] = v
-
-        model = self.model_description[MODEL_BACKEND](**point_mapped)
+        for param, val in point.items():
+            point_mapped[param] = self.space[param][1](val)
+        self.model.set_params(**point_mapped)
 
         try:
-            model.fit(X_train, y_train)
-            r = model.score(X_test, y_test)
+            self.model.fit(X_train, y_train)
+            r = self.model.score(X_test, y_test)
         except BaseException as ex:
             r = 0.0 # on error: return assumed smallest value of objective function
 
         # while negative values could be informative, they could be very large also,
         # which could mess up the optimization procedure. Suggestions are welcome.
-        r = max(r, 0.0)
-
-        return r
+        return max(r, 0.0)
 
 # this is necessary to generate table for README in the end
-table_template = """|Blackbox Function| Minimum | Best minimum | Mean f_calls to min | Std f_calls to min | Fastest f_calls to min
```

Comment:
I removed the f_calls parameter for now (since we are also averaging across different model-dataset combinations.